src/main/java/org/folio/services/transactions/PaymentService.java

```diff
@@ -29,8 +30,20 @@ public PaymentService(TransactionService transactionService) {
   }
 
   @Override
-  public CompletableFuture<Void> updateTransaction(Transaction transaction, RequestContext requestContext) {
-    return unsupportedOperationExceptionFuture();
+  public CompletableFuture<Void> updateTransaction(Transaction payment, RequestContext requestContext) {
+    return FolioVertxCompletableFuture.runAsync(requestContext.getContext(), () -> {
+        transactionService.validateTransactionType(payment, Transaction.TransactionType.PAYMENT);
+        if (!Boolean.TRUE.equals(payment.getInvoiceCancelled()))
+          throw new HttpException(422, UPDATE_PAYMENT_TO_CANCEL_INVOICE.toError());
+      })
+      .thenCompose(v -> transactionService.retrieveTransactionById(payment.getId(), requestContext))
+      .thenAccept(existingTransaction -> {
+        existingTransaction.setInvoiceCancelled(true);
+        existingTransaction.setMetadata(payment.getMetadata());
```

Comment:
Metadata is readonly field and no need to be set intentionally, it will be overwritten by RMB internal mechanisms anyway

Reply:
"setMetadata" is needed here to compare the 2 transactions: otherwise one of them has metadata and the other does not, and the comparison fails because of it. "existingTransaction" is only used temporarily for the comparison, it is not saved. Let me know if you know of a better way to compare only values without taking the metadata into account. I could write a custom method to compare, but that would be a bit heavy...

Reply:
If you need it for unit test, then take a look please: org.folio.rest.impl.EntitiesCrudBasicsTest#compareRecordWithSentToStorage

Reply:
This is not for tests. I am adding a comment to clarify.